community/kernel/src/main/java/org/neo4j/kernel/api/schema_new/constaints/ConstraintDescriptor.java

```diff
@@ -32,8 +32,28 @@
 {
     public enum Type
     {
-        UNIQUE,
-        EXISTS
+        UNIQUE( true, false ),
+        EXISTS( false, true ),
+        UNIQUE_EXISTS( true, true );
```

Comment:
I think `UNIQUE_EXISTS` is descriptive of the behaviour while "key" is an overloaded term, so I think this is fine as is.